core/src/main/scala/kafka/admin/AclCommand.scala

```diff
@@ -124,6 +128,17 @@ object AclCommand {
     }
   }
 
+  def validatePrincipalTypes(resourceToAcl: Map[Resource, Set[Acl]], validPrincipalTypes: Set[String]) = {
+    for((_, acls) <- resourceToAcl) {
```

Comment:
Why are we receiving `resourceToAcl` if we only care about the ACLs? Maybe you want to use that information to provide better error messages? If not, then we should change what we receive to be `Set[Acl]` or something.